backend/molgenis-emx2-datamodels/src/main/java/org/molgenis/emx2/datamodels/profiles/SchemaFromProfile.java

```diff
@@ -59,23 +65,132 @@ public SchemaMetadata create() throws MolgenisException {
+          if (profilesForColumn.contains(profileForTable)) {
+            atLeastOneColumnIsTagged = true;
+            break;
+          }
         }
-        for (String profile : profiles) {
-          if (this.profiles.getProfileTagsList().contains(profile)) {
+        // tag all when none are tagged by this particular profileTag
+        if (!atLeastOneColumnIsTagged) {
+          for (Row column : columnDeclarationsByTableName.get(tableName)) {
+            List<String> profilesForColumn =
+                column.getString("profiles") != null
+                    ? csvStringToList(column.getString("profiles"))
+                    : new ArrayList<>();
+            profilesForColumn.add(profileForTable);
+            column.setString("profiles", String.join(",", profilesForColumn));
+          }
+        }
+      }
+    }
+
+    // post process 2: add non-tagged inherited tables of tagged tables to eliminate the need to tag
+    // them and inadvertently adding all the columns of these parent tables or including 1 random
+    // column to prevent the rest from being added
+    Map<String, String> tableToParent = new HashMap<>();
+    for (String tableName : tableDeclarationsByTableName.keySet()) {
+      Row row = tableDeclarationsByTableName.get(tableName);
+      if (row.getString("tableExtends") != null) {
+        tableToParent.put(tableName, row.getString("tableExtends"));
+      }
+    }
+    for (String tableName : tableDeclarationsByTableName.keySet()) {
+      Row row = tableDeclarationsByTableName.get(tableName);
+      List<String> profilesForTable = csvStringToList(row.getString("profiles"));
+      for (String profileForTable : profilesForTable) {
+        if (this.profiles.getProfileTagsList().contains(profileForTable)
+            && tableToParent.containsKey(tableName)) {
+          // table has a parent, add the current tag to it and continue to its parent and so on
+          String nextTable = tableName;
+          while (tableToParent.containsKey(nextTable)) {
+            String parentTableName = tableToParent.get(nextTable);
+            Row parentTable = tableDeclarationsByTableName.get(parentTableName);
+            List<String> updateProfilesForParentTable =
+                parentTable.getString("profiles") != null
+                    ? csvStringToList(parentTable.getString("profiles"))
+                    : new ArrayList<>();
+            if (!updateProfilesForParentTable.contains(profileForTable)) {
+              updateProfilesForParentTable.add(profileForTable);
+              parentTable.setString("profiles", String.join(",", updateProfilesForParentTable));
+            }
+            nextTable = parentTableName;
+          }
+        }
+      }
+    }
+
+    // finally, run profile select as usual, first per table
+    for (String tableName : tableDeclarationsByTableName.keySet()) {
+      Row row = tableDeclarationsByTableName.get(tableName);
```

Comment:
I would have expected you would here have checked for the 'table level' profile and then add all columns instead of first changing the column tags.